pagespeed/system/controller_manager.cc

```diff
@@ -0,0 +1,335 @@
+// Copyright 2016 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// Author: jefftk@google.com (Jeff Kaufman)
+
+#include "pagespeed/system/controller_manager.h"
+
+#include <signal.h>
+#include <sys/wait.h>
+#include <unistd.h>
+
+#include "pagespeed/system/system_rewrite_options.h"
+
+namespace net_instaweb {
+
+
+ControllerManager::ControllerManager(
+    SystemRewriteDriverFactory* factory,
+    ThreadSystem* thread_system,
+    FileSystem* file_system,
+    Timer* timer,
+    MessageHandler* handler) : factory_(factory),
+                               thread_system_(thread_system),
+                               file_system_(file_system),
+                               timer_(timer),
+                               handler_(handler),
+                               babysitter_pid_(-1) { }
+
+void ControllerManager::SetUpSignalHandlers() {
+  // We need to clear inherited signal handlers. There's no portable way to get
+  // a list of all possible signals, but these are the rest of the ones that we
```

Comment:
we can get this data from `signal.h`, and use the macro for `NSIG`.

Since the signal numbers are allocated consecutively, `NSIG` is also one greater than the largest defined signal number.